### ISSUE TITLE
Update Components Variants article to align with V22 components changes

### ISSUE THREAD

**Issue by web-padawan** [vaadin/docs#882]
## Description

The [Variant Inheritance](
https://github.com/vaadin/docs/blame/e2d86ad426f730f6ddb81b009e65816834db3c7e/articles/ds/customization/component-variants.asciidoc#L53-L54) is no longer up to date and needs to be updated:

```
For example, <<../components/combo-box#, Combo Box>> contains a <<../components/text-field#, Text Field>> in its shadow DOM, so theme variants applied to a Combo Box are automatically passed down to the Text Field inside.
This means that you can apply Text Field variants to a Combo Box even though Combo Box doesn't have any variants of its own.
```
